[itsmejr257] iP

itsmejr257 opened the pull request.

itsmejr257 committed:
Rename Duke to JunBot

itsmejr257 committed:
Remove code for Duke

itsmejr257 committed:
Add greetings

itsmejr257 committed:
Add exit messeges

itsmejr257 committed:
Add command echoing and exit on 'bye' command

This commit introduces a new feature that echoes commands entered by
user in the command-line. Application exits when user types 'bye'

itsmejr257 committed:
Rename echoCommands function to commandHandler

Rename function to better describe its functionality

itsmejr257 committed:
Add addToList() and printList() functions

Add the ability to store whatever text entered by the user and display
them back to the user when requested.

itsmejr257 committed:
Add Task Class

Create a Task class to represent tasks

itsmejr257 committed:
Refactor JunBot to use Task

itsmejr257 committed:
Add mark, unmark and print functionality

This commit implements ability to mark and unmark tasks, along with a
new feature to print tasks with its status icon.

itsmejr257 committed:
Apply Google Java Style Guide and CS2113 Java Coding Standard
to JunBot.java and Task.java

Follow consistent coding practices by standardizing indentation,
spacing, and naming conventions

itsmejr257 committed:
Add Todo, Event and Deadlines functions

Add todo class for user to add task without dates. Add event class for
user to add task with start and end dates. Add deadlines class for
user to add task with end date.

itsmejr257 committed:
Add automated text ui testing

itsmejr257 committed:
Refactor mark and unmark function to use isValidListPosition function

Add a common Boolean function to determine if user input
has a valid list number for mark and unmark functions

Ijaaz01 commented on src/main/java/JunBot.java:
```diff
@@ -59,18 +59,48 @@ public static void addTodo(String description){
+    public static void unmarkTaskInList(String command) {
         command = command.replace("unmark", "").trim();
+
+        if (!isValidListPosition(command)){
```
I like that you used the 'is' prefix for a boolean method, good job!

itsmejr257 replied:
Thank you!

Ijaaz01 commented on src/main/java/JunBot.java:
```diff
@@ -59,18 +59,48 @@ public static void addTodo(String description){
         int listNumber = Integer.parseInt(command) - 1;
         tasks[listNumber].markTask();
+
+        System.out.println(DIVIDER + "Nice! I've marked this task as done:\n");
```
I like the use of all caps for the static string divider, well done!

itsmejr257 replied:
Thank you!

Ijaaz01 commented on src/main/java/Event.java:
```diff
@@ -1,15 +1,15 @@
         this.endDate = endDate;
     }
 
     @Override
```
Well done on using '@Override' to indicate the override makes it easier and faster to understand the purpose of the method!

itsmejr257 replied:
Thank you!

Ijaaz01 commented on src/main/java/Deadline.java:
```diff
@@ -1,14 +1,14 @@
 
     @Override
-    public String toString(){
+    public String toString() {
```
Good job on adding the spacing after the method name!

itsmejr257 replied:
Thank you!

Ijaaz01 reviewed (commented):
Great code quality overall keep t up!

itsmejr257 committed:
Update EXPECTED.TXT for text ui tests

itsmejr257 committed:
Add InvalidInputException class

Create a class to represent exceptions specific to inputs

itsmejr257 committed:
Refactor JunBot to use InvalidInputException for error handling

Add error handling for invalid inputs for todo, deadline and event.

itsmejr257 committed:
Merge branch 'branch-Level-5'

itsmejr257 committed:
Divide classes into packages

Organize classes into multiple packages to distinguish between
commands and task

itsmejr257 committed:
Add error package

itsmejr257 committed:
Merge branch 'branch-A-Packages'

itsmejr257 committed:
Refactor tasks array to use ArrayList<Task>

itsmejr257 committed:
Add deleteTask function to JunBot

Add support for deleting tasks from the list

itsmejr257 committed:
Add Save function to JunBot

Add ability to save tasks in hard disk and load data from the
hard disk when chatbot starts

itsmejr257 committed:
Merge branch 'branch-Level-6'

itsmejr257 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/JunBot.java

itsmejr257 committed:
Add error handling if tasks.txt does not exist

Add error handing to create relevant directory and file if path does
not exist

aditichadha1310 commented on src/main/java/JunBot.java:
```diff
@@ -0,0 +1,291 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class JunBot {
```
Comments are missing. Good to add them for better readability.

aditichadha1310 commented on src/main/java/JunBot.java:
```diff
@@ -0,0 +1,291 @@
+            return;
+        }
+
+        int listNumber = Integer.parseInt(command) - 1;
```
Avoid the usage of Magic numbers.

aditichadha1310 commented on src/main/java/JunBot.java:
```diff
@@ -0,0 +1,291 @@
+
+    }
+
+    public static void handleUserInput() throws IOException {
```
Avoid long methods. Take corrective action if method exceeds 30 lines of code.

aditichadha1310 reviewed (commented):
Overall, Good job!

itsmejr257 committed:
Add Ui, Storage, Parser and TaskList classes

Add classes to extract out closely related code

itsmejr257 committed:
Add Dates and Times to deadline and event commands

Add dates and times format to deadline and event commands

itsmejr257 committed:
Add find functionality to TaskList

Add a way for users to find a task by searching for a keyword

itsmejr257 committed:
Add JavaDoc comments

Add JavaDoc comments to the code

itsmejr257 committed:
Merge pull request #1 from itsmejr257/branch-Level-8

Add Dates and Times to deadline and event commands

itsmejr257 committed:
Merge branch 'master' of https://github.com/itsmejr257/ip

itsmejr257 committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	data/tasks.txt
#	src/main/java/junbot/common/TaskList.java

itsmejr257 committed:
Merge pull request #2 from itsmejr257/branch-Level-9

Add find functionality to TaskList

itsmejr257 committed:
Merge branch 'master' of https://github.com/itsmejr257/ip

itsmejr257 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/JunBot.java
#	src/main/java/junbot/common/TaskList.java

itsmejr257 committed:
Merge pull request #3 from itsmejr257/branch-A-JavaDoc

Add JavaDoc comments

itsmejr257 committed:
Merge branch 'master' of https://github.com/itsmejr257/ip

itsmejr257 committed:
Fix null pointer exception in Storage.java

Fix null pointer exception when marking tasks due to incorrect
initialization value of counter

itsmejr257 committed:
Fix find function to display correct number of tasks

Fix find function not displaying tasks if there is only 1 found task.
This is due to incorrect handling of the for loop range.

itsmejr257 committed:
Add JunBot user guide to README

itsmejr257 committed:
Add JunBot user guide to README in docs folder

itsmejr257 committed:
Refactor code to adhere to Java coding conventions

itsmejr257 committed:
Update README.md formatting for better rendering on Github Pages

itsmejr257 committed:
Refactor README.md for improved compatibility with GitHub Pages

itsmejr257 committed:
Remove link for JunBot.jar from README.md

itsmejr257 committed:
Refactor TaskList to follow Java Coding Conventions

itsmejr257 committed:
Refactor Task to remove unused function printTask()

itsmejr257 committed:
Refactor comments in TaskList to match correct parameter name

itsmejr257 committed:
Refactor JunBot.java to remove unnecessary exceptions

Remove unused IOException in findTasks and unused InvalidInputException
in main

itsmejr257 committed:
Add JavaDoc comments

itsmejr257 committed:
Update user guide to specify DATE format
